src/sqlfluff/rules/L010.py

```diff
@@ -78,6 +78,15 @@ def _eval(self, context: RuleContext) -> Optional[List[LintResult]]:
         if self.matches_target_tuples(context.segment, self._exclude_elements, parent):
             return [LintResult(memory=context.memory)]
 
+        # If the parent element is a function_name and there is more than one element
+        # below it, we ignore it because it is most likely a user defined function.
+        # SUM() should be modified. However, dbo.myScalar() must not be modified.
```

Comment:
I found this difficult to follow. Would it be easier to just say something like this:

```suggestion
        # Used by L030 (that inherits from this rule)
        # If it's a qualified function_name (i.e with more than one part to
        # function_name). Then it is likely an existing user defined function (UDF)
        # which are case sensitive so ignore for this.
```

I wonder should there be a `include_qualified_functions` option in this (defaulted to `false`)?

Also, I think in some SQL dialects there might be a `SYS` schema or such like that contains the system functions, which are case insensitive? Then again not sure if there's a good reason to use that, so maybe fine to ignore that edge case.